### ISSUE TITLE
Too hard to help this project

### ISSUE THREAD

**Issue by flaviojs** [ja2-stracciatella/free-ja2-resources#1]
I don't have the original editor.slf so I can't use `create_free_editorslf.py` to build a free editor.slf.

With programming knowledge:
I can see which STI files are used by searching for EDITORDIR in ja2-stracciatella.
I can see which images are needed by following into the code that uses EDITORDIR.
But I don't know the size that each image should have.

Even if I had the original editor.slf:
There is no apparent way to convert the STIs to GIFs, to use as reference for the new images.
There is no mention that the gif images in the same STI need to have the same palette (if I understood `create_free_editorslf.py` correctly).

In short, it is too difficult for a random artist to contribute to this project.
I propose making a list with the files that are needed, their purpose, and image size.

**Comment by flaviojs** [comment]
This is what I got from inspecting the ja2-stracciatella source: [editor.txt](https://github.com/ja2-stracciatella/free-ja2-resources/files/3192400/editor.txt)
If someone can complement with image sizes, that would be great. ;)

**Comment by selaux** [comment]
Here you go [sizes.txt](https://github.com/ja2-stracciatella/free-ja2-resources/files/3193827/sizes.txt).

**Comment by flaviojs** [comment]
I can't control the palette when exporting to gif/png with GIMP.
Do you know of a free image editor for linux or in the web that gives you control of the palette?

Another alternative is to make `create_free_editorslf.py` handle different palettes gracefully.

**Comment by selaux** [comment]
I used GIMP as well AFAIR. But I had a dump of the original `editor.slf` available, so I just used the included palettes to keep the style of the original game. Basically in GIMP you then only need to open the file go `Windows -> Dockable Dialogs -> Palettes` and then there is `Colormap of image #1`, this one you can edit which opens another dialog where you can pick and change colors. Idk if there is a better way as I'm not an artist.

Here is a dump of editor slf with all images included with their respective palette, but without the actual image (just everything set to one color).
[editor.zip](https://github.com/ja2-stracciatella/free-ja2-resources/files/3194876/editor.zip)

**Comment by flaviojs** [comment]
I was able to see the palette and exporting preserved the palette. (GIMP)
But I wasn't able to change palette colors, so it's a limited solution.

However, I though of a different way.
The problem is how to make them share the same palette so how about accepting a single image with all subimages?
The problem turns into "which section of the image belongs to each subimage", and an accompanying file in a yet-to-be-determined format can tell that to the script.

**Comment by selaux** [comment]
Would be fine for me as well.

**Comment by flaviojs** [comment]
I'm thinking of something like this: ARROWSOFF.json
```json
{
  "comment": "example for ARROWSOFF.STI which contains images for the merc direction buttons",
  "image": "ARROWSOFF.gif",
  "subimages": [
    { "offset": { "x":   0, "y": 0}, "size": { "x": 29, "y": 29}, "comment": "0=northwest" },
    { "offset": { "x":  29, "y": 0}, "size": { "x": 30, "y": 30}, "comment": "1=north" },
    { "offset": { "x":  59, "y": 0}, "size": { "x": 29, "y": 29}, "comment": "2=northeast" },
    { "offset": { "x":  88, "y": 0}, "size": { "x": 30, "y": 30}, "comment": "3=east" },
    { "offset": { "x": 118, "y": 0}, "size": { "x": 29, "y": 29}, "comment": "4=southeast"},
    { "offset": { "x": 147, "y": 0}, "size": { "x": 30, "y": 30}, "comment": "5=south" },
    { "offset": { "x": 177, "y": 0}, "size": { "x": 29, "y": 29}, "comment": "6=southwest" },
    { "offset": { "x": 206, "y": 0}, "size": { "x": 30, "y": 30}, "comment": "7=west" },
    { "image": "ARROWSOFF.STI/8.gif", "comment": "example image as extra subimage" },
    { "image": "static.gif", "offset": { "x": 0, "y": 0 }, "size": { "x": 30, "y": 30 }, "comment": "you can take just part of the image" },
  ]
}
```

**Comment by flaviojs** [comment]
With #2 and #3 merged, my concerns are handled.
Docs still need improving but that's not something I'll work on.
I think this can be closed?

**Comment by lynxlynxlynx** [comment]
Sure, it's also cleaner this way.
